File: client/app/services/aiBuilderService.ts
import { apiClient } from '../lib/api-client';
import { API_ENDPOINTS } from '../lib/config';

export const AIBuilderService = {
  async generateWorkflow(
    question: string,
    options: {
      credentialId: string;
      modelName: string;
      baseUrl?: string;
      mode?: 'build' | 'edit';
      existingWorkflow?: { nodes: any[]; edges: any[] };
    }
  ) {
    const response = await apiClient.post<any>(API_ENDPOINTS.AI_BUILDER.GENERATE, {
      question,
      credential_id: options.credentialId,
      model_name: options.modelName,
      base_url: options.baseUrl || undefined,
      mode: options.mode || 'build',
      existing_workflow: options.existingWorkflow || null,
    });
    return response;
  }
};

File: backend/app/services/ai_builder_service.py
import logging
from typing import Dict, Any
from app.services.ai_builder.ai_builder_graph import generate_workflow as core_generate_workflow

logger = logging.getLogger(__name__)

class AIBuilderService:
    """
    Facade for the AI Builder service, wrapping the new LangGraph modular implementation.
    This maintains backward compatibility for any existing code calling AIBuilderService.
    """
    def __init__(self):
        # Dependencies like OpenAI API key and Validation engines are now managed
        # directly within their respective modular processors.
        pass

    async def generate_workflow(self, question: str, api_key: str, model_name: str = "gpt-4o", base_url: str = None, mode: str = "build", existing_workflow: Dict[str, Any] = None) -> Dict[str, Any]:
        """
        Delegates the AI workflow building to the modular LangGraph package.
        """
        logger.info(f"AIBuilderService wrapper called for mode: {mode}")
        return await core_generate_workflow(question, api_key, model_name, base_url, mode, existing_workflow)
